### PR TITLE
device: using const strings for block-driver option instead of hard coding

### PR DESCRIPTION
**Pull request by Kvasscn**
Currently, the block driver option is specifed by hard coding, maybe it
is better to use const string variables instead of hard coded strings.

Fixes: #3321

Signed-off-by: zhanghj <zhanghj.lc@inspur.com>

### CODE REVIEW

File: src/runtime/virtcontainers/device/drivers/block.go
Comment (liubin) [inline src/runtime/virtcontainers/device/drivers/block.go:22]: Non-package public const need not be started with uppercase letters.

Comment (Kvasscn) [reply src/runtime/virtcontainers/device/drivers/block.go]: @liubin thanks, already commit again.

**Review by amshinde — DISMISSED** [review]

**Comment by gkurz** [comment]
The intent is good but  both `"fstype"` and `"block-driver"` have users in other files.

```
src/runtime/virtcontainers/device/config/config.go:     // for example, for BlockDevice, we can set DriverOptions["blockDriver"]="virtio-blk"
src/runtime/virtcontainers/device/config/pmem.go:       device.DriverOptions["fstype"] = fstype
src/runtime/virtcontainers/device/manager/manager.go:           devInfo.DriverOptions["block-driver"] = dm.blockDriver
src/runtime/virtcontainers/device/manager/manager.go:           devInfo.DriverOptions["block-driver"] = dm.blockDriver
src/runtime/virtcontainers/documentation/api/1.0/api.md:        // for example, for BlockDevice, we can set DriverOptions["blockDriver"]="virtio-blk"
src/runtime/virtcontainers/documentation/api/1.0/api.md:        // for example, for BlockDevice, we can set DriverOptions["blockDriver"]="virtio-blk"
src/runtime/virtcontainers/persist/api/device.go:       // for example, for BlockDevice, we can set DriverOptions["blockDriver"]="virtio-blk"
```

I suggest you define these strings in a more common file (`config/config.go` ?) and maybe fix the other users as well, while here.

**Review by jodh-intel — APPROVED** [review]
Thanks @Kvasscn.

lgtm

**Comment by jodh-intel** [comment]
Good catch @gkurz - added `do-not-merge` until that's also handled.

**Review by gkurz — CHANGES_REQUESTED** [review]
Looks overall good to me. Please address the comments and this should be ready to go.

Side question : [src/runtime/virtcontainers/device/manager/manager.go](https://github.com/kata-containers/kata-containers/blob/e6e5d2593ac319329269d7b58c30f99ba7b2bf5a/src/runtime/virtcontainers/device/manager/manager.go#L24) defines the exact same strings for the block driver names as [src/runtime/virtcontainers/device/config/config.go](https://github.com/kata-containers/kata-containers/blob/e6e5d2593ac319329269d7b58c30f99ba7b2bf5a/src/runtime/virtcontainers/device/config/config.go#L24). Is there any reason to keep the duplication ?

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

**Comment by jodh-intel** [comment]
Any update on this @Kvasscn?

**Comment by Kvasscn** [comment]
Already updated the PR， please  review  again. @gkurz  @jodh-intel   
Thanks.

File: src/runtime/virtcontainers/device/config/config.go
Comment (Kvasscn) [reply src/runtime/virtcontainers/device/config/config.go:107]: Already remove the duplicated  definitions for virtio driver in manager.go,  and the following files need to be updated  at same time:  @gkurz 
src/runtime/virtcontainers/container_test.go
src/runtime/virtcontainers/kata_agent_test.go
src/runtime/virtcontainers/persist_test.go
src/runtime/virtcontainers/sandbox_test.go

**Review by jodh-intel — APPROVED** [review]
Thanks @Kvasscn - nice cleanup!

lgtm

**Review by gkurz — APPROVED** [review]
Sorry for the delay... All comments were addressed. Thanks @Kvasscn !

**Comment by gkurz** [comment]
@Kvasscn, there's a conflict that needs to be addressed before we can merge though. PTAL.

**Comment by jodh-intel** [comment]
Any update on this @Kvasscn ?

**Comment by Kvasscn** [comment]
just resolved merge conflicts.  @jodh-intel

**Review by gkurz — CHANGES_REQUESTED** [review]
Hi @Kvasscn !

We're getting close. I'll ask just one last change and we should be good. My apologies for not spotting that before.

DCO compliance requires your Signed-off-by: tag to be [using your real name (sorry, no pseudonyms or anonymous contributions.)](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/process/submitting-patches.rst?id=HEAD#n407).

In your case this should thus be:

```
Signed-off-by: Jason Zhang <zhanghj.lc@inspur.com>
```

References:
https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#certificate-of-origin
https://ltsi.linuxfoundation.org/software/signed-off-process/
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/process/submitting-patches.rst?id=HEAD#n403

**Comment by jodh-intel** [comment]
Thanks @Kvasscn. @gkurz - can you tal please?

**Comment by Kvasscn** [comment]
Already updated signed-off-by tag  in the commit, thanks  @gkurz

**Review by gkurz — APPROVED** [review]
Thanks @Kvasscn !

**Comment by liubin** [comment]
@jodh-intel the `do-not-merged` label is added by you, so I think it's better to remove it yourself and go on.

**Comment by jodh-intel** [comment]
Thanks @Kvasscn!

**Pull request merged**
